pkg/planner/core/casetest/partition/testdata/integration_partition_suite_out.json

```diff
@@ -1440,7 +1440,7 @@
       {
         "SQL": "select * from tlist2 where a = 1 and b in (1,2) order by b",
         "DynamicPlan": [
-          "TableReader 2.00 root partition:p0,p1 data:TableRangeScan",
+          "TableReader 2.00 root partition:all data:TableRangeScan",
```

Comment:
Will this PR cause some performance regression since I see in some cases that partition pruning can work well as before?

Reply:
This difference is just making it more clear, that `p0,p1` is actually `all` partitions, so there should be no performance regression related to partition pruning.
The impact in general should be minimal, it will need to check if the table's PartitionInfo is in a state that can have overlapping partitions (very rare to hit, only during one State change for DROP PARTITION), and if it is, then some smaller CPU usage for handling the partitions.